### PR TITLE
add vAppConfig as a new VirtualMachineMetadataTransport

### PR DESCRIPTION
**Pull request by zyiyi11**
This change adda s new `VirtualMachineMetadataTransport` called `vAppConfig`. 
This will set vAppConfig properties for VMs imported through OVFs. It will allow images bootstrap themselves with OVF properties that define the network information instead of relying on Guest OS Customization (GOSC).

### CODE REVIEW

File: api/v1alpha1/virtualmachine_types.go [hunk 1]
```diff
@@ -129,6 +129,12 @@ const (
 	// or as vApp properties on an existing VM or VMTX will be set, all others will be ignored.
 	VirtualMachineMetadataOvfEnvTransport VirtualMachineMetadataTransport = "OvfEnv"
 
+	// VirtualMachineMetadataVAppConfigTransport indicates that the data set in the VirtualMachineMetadata Transport Resource,
+	// i.e., a ConfigMap or Secret, will be vApp properties on the VM, which will be exposed as vAppConfig to the Guest VM. Only properties
+	// marked userConfigurable and already present in either OVF Properties of a VirtualMachineImage
+	// or as vApp properties on an existing VM or VMTX will be set, all others will be ignored.
+	VirtualMachineMetadataVAppConfigTransport VirtualMachineMetadataTransport = "vAppConfig"
```
Comment (akutz) [inline api/v1alpha1/virtualmachine_types.go:136]: A few things:

* Please try not to exceed 80 columns for comments
* Please add a note that indicates this transport does not use Guest OS customization for networking, and this transport should only be selected if the image exposes OVF properties that bootstrap the guest's networking

**Review by akutz — CHANGES_REQUESTED** [review]
Thanks @zyiyi11 , looks good aside from two nits.

File: api/v1alpha1/virtualmachine_types.go (hunk 1 above)
Comment (zyiyi11) [reply api/v1alpha1/virtualmachine_types.go:136]: Thanks! I also made changes to all other comments on metadata transport to accommodate 80 columns.

Comment (akutz) [reply api/v1alpha1/virtualmachine_types.go:146]: Ha, thanks! I had done the same in a WIP branch for v1a2 which is why I just asked you to handle yours, but I should have known you would have not stopped at that :)

Comment (zyiyi11) [reply api/v1alpha1/virtualmachine_types.go:146]: haha yeah. I can't stand that they are not aligned. ;)

File: api/v1alpha1/virtualmachine_types.go [hunk 2]
```diff
@@ -113,22 +113,38 @@ type VirtualMachineNetworkInterface struct {
...
 	VirtualMachineMetadataOvfEnvTransport VirtualMachineMetadataTransport = "OvfEnv"
 
+	// VirtualMachineMetadataVAppConfigTransport indicates that the data set in
+	// the VirtualMachineMetadata Transport Resource, i.e., a ConfigMap or Secret,
+	// will be vApp properties on the VM, which will be exposed as vAppConfig to the Guest VM.
+	// Only properties marked userConfigurable and already present in either
+	// OVF Properties of a VirtualMachineImage or as vApp properties on an existing VM
+	// or VMTX will be set, all others will be ignored.
+	//
+	// This transport should only be selected if the image exposes OVF properties
```
Comment (akutz) [inline api/v1alpha1/virtualmachine_types.go:144]: Maybe:

```go
// Selecting this transport means the guest's network is not automatically
// configured by vm-tools. This transport should be selected if the image
// exposes OVF/vApp properties that are used by the guest to bootstrap
// its networking configuration.
```

**Review by akutz — APPROVED** [review]

**Pull request merged**